mrmustard/lab_dev/states/states.py

```diff
@@ -371,6 +372,54 @@ def representation(self) -> Bargmann:
         return Bargmann(*triples.vacuum_state_Abc(n_modes))
 
 
+class QuadratureEigenstate(Ket):
+    r"""
+    The `N`-mode Quadrature eigenstate .
+
+    .. code-block ::
+
+        >>> from mrmustard.lab_dev import QuadratureEigenstate
+
+        >>> state = QuadratureEigenstate([1, 2])
+        >>> assert state.modes == [1, 2]
+
+    Args:
+        modes: A list of modes.
+        x: the displacement of the state
+        phi: the angle of the state with `0` being a position eigenstate and `\pi/2` being the momentum eigenstate.
+
+    .. details::
+        Its ``(A,b,c)`` triple is given by
+
+        .. math::
+            A = -I_{N}\exp(i2\phi)\text{, }b = I_Nx\exp(i\phi)\sqrt{2/\hbar}\text{, and }c = 1/(\pi\hbar)^{-1/4}\exp(-\abs{x}^2/(2\hbar)).
+    """
+
+    def __init__(
+        self,
+        modes: Sequence[int],
+        x: Union[float, Sequence[float]] = 0.0,
+        phi: Union[float, Sequence[float]] = 0.0,
+        x_trainable: bool = False,
+        phi_trainable: bool = False,
+        x_bounds: Tuple[Optional[float], Optional[float]] = (None, None),
+        phi_bounds: Tuple[Optional[float], Optional[float]] = (None, None),
+    ):
+        super().__init__(modes=modes, name="QuadratureEigenstate")
+        self._add_parameter(make_parameter(x_trainable, x, "x", x_bounds))
+        self._add_parameter(make_parameter(phi_trainable, phi, "phi", phi_bounds))
+
+    @property
+    def representation(self) -> Bargmann:
+        n_modes = len(self.modes)
+        xs, phis = list(reshape_params(n_modes, x=self.x.value, phi=self.phi.value))
+        return Bargmann(*triples.quadrature_eigenstates_Abc(xs, phis))
+
+    @property
+    def probability(self) -> float:
+        return math.inf
```

Comment:
what happens if we don't say the prob is infinite?

Reply:
The tests still run. It's something Jacob added. I guess it may be useful when calculating norms.

Reply:
Ok. I'm not convinced it's the right thing to do, though. I'd like to first see where problems arise if we don't override this.